src/type/s2k/index.js

```diff
@@ -0,0 +1,35 @@
+import defaultConfig from '../../config';
+import Argon2S2k, { Argon2OutOfMemoryError } from './argon2';
+import OtherS2K from './generic';
+import enums from '../../enums';
+
+const allowedDefaultS2KTypes = new Set([enums.s2k.argon2, enums.s2k.iterated]);
+
+/**
+ * Instantiate a new S2K instance of the given type
+ * @param {module:enums.s2k} [type] - If not specified, `config.s2kTypes` is used
+ * @oaram {Object} [config]
+ * @returns {Object} New s2k object
+ * @throws {Error} for unknown or unsupported types
+ */
+export function newS2KFromType(maybeType, config = defaultConfig) {
+  if (!maybeType && !allowedDefaultS2KTypes.has(config.s2kType)) {
+    throw new Error('The provided `config.s2kType` value is not allowed');
+  }
+
+  const type = maybeType || config.s2kType;
```

Comment:
IMO, this API is a bit strange in the `newS2KFromType(undefined, config)` case. I would prefer to split those cases up, i.e. you could have `newS2KFromConfig(config)`, which then calls `newS2KFromType(type)`.